src/main/java/org/json/JSONObject.java

```diff
@@ -2629,8 +2632,34 @@ public static Object wrap(Object object) {
         return wrap(object, null);
     }
 
+    // Custom isFiniteNumber method
+    public static boolean isFiniteNumber(Object value) {
+        if (value instanceof Number) {
+            double doubleValue;
+            if (value instanceof Integer) {
+                doubleValue = ((Integer) value).doubleValue();
```

Comment:
Integers are always finite.

Reply:
agreed. Using the existing `testValidity` method would be the correct course and already covers what this PR is attempting.

Reply:
agree @johnjaylward  the testValidity can alone verify this, am somehow new to this code base, can you help me understand how did you found this method.